Automattic/newspack-plugin
# feat: update RAS settings layout to i3 designs

### All Submissions:






### Changes proposed in this Pull Request:



Updates the design of the new RAS settings page to the latest i3 designs. This PR covers only the first settings page in **Newspack > Engagement > Reader Activation**. The setup wizard for Campaigns defaults will come in a future PR.

<img width="1205" alt="Screen Shot 2023-04-03 at 4 04 29 PM" src="https://user-images.githubusercontent.com/2230142/229637491-378ad7a9-0092-4729-8129-112b23ebebce.png">

### How to test the changes in this Pull Request:

1. Start with a fresh test site (JN). Install the plugin from this branch and run `wp newspack setup` to scaffold the site with default Newspack settings.
2. Visit **Newspack > Engagement > Reader Activation** to view the checklist of prerequisites that need to be completed before RAS Campaigns defaults can be configured.

<img width="1065" alt="Screen Shot 2023-04-03 at 4 11 36 PM" src="https://user-images.githubusercontent.com/2230142/229638543-5b810818-ba94-4173-b0d4-252cf2adda9d.png">

3. Expand each card and complete the setup.
  - For cards that contain fields and a "Save settings" button, fill out the required fields and click "Save" (note that some fields will be prepopulated with suggested settings, but you still need to click "save" to complete the prerequisite). After saving, confirm that a.) the card collapses automatically with a "status: ready" state, and b.) if you re-expand the card, the button label changes to say "Update settings" instead of "Save settings."
  - For cards that link out to a separate page to complete configuration, complete the required config and then go back to the **Newspack > Engagement > Reader Activation** page. Confirm that the card updates to say "status: ready" after completing the required configuration.
4. The last card ("Reader Activation Campaign") is just a placeholder for now. For this PR, just confirm that the action button is disabled with a "Waiting for all settings to be ready" label until all other settings are ready, after which the button should change to an active "Set up Reader Activation campaign" button which currently does nothing but will redirect to the prompt setup wizard in a future PR.
5. Also confirm that the "Complete the settings to enable Reader Activation" banner at the top of the page disappears after you complete all steps (in a future PR this will happen only after you complete the not-yet-existent prompt setup wizard).
6. Expand the Advanced Settings panel and confirm that the "Save advanced settings" button now only saves changes from the fields in advanced settings, not elsewhere on the page.

### Other information:
